src/types/abstract_hmm.jl

```diff
@@ -66,18 +89,32 @@ Return the vector of initial state log-probabilities for `hmm`.
 Falls back on `initialization`.
 """
 log_initialization(hmm::AbstractHMM) = elementwise_log(initialization(hmm))
+	
+log_initialization(hmm::AbstractHMM, control) = elementwise_log(initialization(hmm, control))
+	
+log_initialization(hmm::AbstractHMM, ::Nothing) = elementwise_log(initialization(hmm))
 
+	
+#------------------------- 2.5. transition_matrix -------------------------#
 """
     transition_matrix(hmm)
     transition_matrix(hmm, control)
 
 Return the matrix of state transition probabilities for `hmm` (possibly when `control` is applied).
 
 !!! note
-    When processing sequences, the control at time `t` influences the transition from time `t` to `t+1` (and not from time `t-1` to `t`).
+    When processing sequences, the control at time `t` influences the transition 	from time `t` to `t+1` (and not from time `t-1` to `t`).
 """
 function transition_matrix end
 
+transition_matrix(hmm::AbstractHMM) = hmm.trans
+
+transition_matrix(hmm::AbstractHMM, control) = transition_matrix(hmm)
+
+transition_matrix(hmm::AbstractHMM, ::Nothing) = transition_matrix(hmm)
```

Comment:
Same remark about assuming existence of fields in an instance of `AbstractHMM`

Reply:
OK, so, just to check if I understood it correctly: the `transition_matrix(hmm::AbstractHMM) = hmm.trans` line is out, as it supposes the field `trans`. The other lines are fallbacks, so they're fine and I can keep them here? Is that it?

My main intention was to move the fallbacks to whatever section their primary function was, instead of keeping them together. I ended up overdoing it and adding the function, sorry!

Reply:
You can keep the fallbacks indeed, and I'd rather have them all in the same file since the other fallbacks are already there